.github/workflows/smoke-test/smoke-test.sh

```diff
@@ -0,0 +1,223 @@
+# first argument).
+win32_verify() {
+    local path
+    path="$(cygpath --windows "$1")"
+    # When running GitHub actions, using `powershell.exe` here causes issues
+    # with loading the `Microsoft.PowerShell.Security` module; using `pwsh.exe`
+    # seems to be fine.  This is probably because the default shell is pwsh, and
+    # the environment has paths to the PowerShell 7 version of the module, so it
+    # tries to load that instead of the version appropriate for PowerShell.exe.
+    local pwsh=(pwsh.exe -NoLogo -NoProfile -NonInteractive -Command)
+    local stdout
+    stdout=$("${pwsh[@]}" "\$(Get-AuthenticodeSignature '$path').Status")
+
+    if [[ "$stdout" != "Valid" ]]; then
+        printf "%s is not correctly signed:\n" "$path"
+        "${pwsh[@]}" "Get-AuthenticodeSignature '$path' | Format-List"
+        exit 1
+    fi
+}
+
+# Assume the first argument given is a path to the Rancher Desktop installer;
+# install it, and set the global variable RDCTL to the path of the rdctl
+# executable.
+install_win32() {
+    local archiveName=$1
+
+    win32_verify "$archiveName"
+    mkdir -p "$(cygpath --unix "${RD_LOGS_DIR}")"
+    MSYS2_ARG_CONV_EXCL='*' msiexec.exe '/lv*x' "${RD_LOGS_DIR}\\install.log" \
+        /i "$(cygpath --windows "$archiveName")" /passive ALLUSERS=1
+    # msiexec returns immediately and runs in the background; wait for that
+    # process to exit before continuing.
+    local deadline
+    deadline=$(( $(date +%s) + 10 * 60 ))
+    while [[ $(date +%s) -lt $deadline ]]; do
+        if MSYS2_ARG_CONV_EXCL='*' tasklist.exe /FI "ImageName eq msiexec.exe" | grep msiexec; then
+            break
+        fi
+        printf "Waiting for msiexec: %s/%s\n" "$(date)" "$(date --date="@$deadline")"
+        sleep 10
+    done
+    local installDirectory
+    installDirectory=$(cygpath --unix 'C:\Program Files\Rancher Desktop')
+    local rdctl="$installDirectory/resources/resources/win32/bin/rdctl.exe"
+
+    local -a keys
+    mapfile -t keys < <(yq.exe 'keys | .[]' < build/signing-config-win.yaml)
+    local key
+    for key in "${keys[@]}"; do
+        local expr='.[env(key)][] | select(. != "!*")'
+        local -a values
+        mapfile -t values < <(key=$key yq.exe "$expr" < build/signing-config-win.yaml)
+        for value in "${values[@]}"; do
+            if [[ "$value" == "wix-custom-action.dll" ]]; then
+                # This file is not installed
+                continue
+            fi
+            win32_verify "$installDirectory/$key/$value"
+        done
+    done
+
+    # Verify that rdctl exists
+    win32_verify "$rdctl"
+    RDCTL=$rdctl
+}
+
+# Wait for the backend to be alive.  $RDCTL must be set (from the install_*
+# functions).
+wait_for_backend() {
+    local deadline state
+    deadline=$(( $(date +%s) + 10 * 60 ))
+
+    while [[ $(date +%s) -lt $deadline ]]; do
+        state=$(MSYS2_ARG_CONV_EXCL='*' "$RDCTL" api /v1/backend_state || echo '{"vmState": "UNREADY"}')
```

Comment:
Is there any reason not to set this once on script start instead of prefixing individual commands with it?

```shell
export MSYS2_ARG_CONV_EXCL='*'
```

At first I thought this line was Win32-specific until I realized that the variable will just be ignored on other platforms.